src/main/java/com/gildedrose/DecreaseBy2.java

```diff
@@ -0,0 +1,7 @@
+package com.gildedrose;
+
+public class DecreaseBy2 implements Updatable {
```

Comment:
Suggestion/Blocker: I think this abstraction goes too far. Seeing classes that only augment one property, and do it in such a specific, single-use way, is a code-smell and tells me at least this part of the pattern is not the correct solution. For example, if we kept adding items that had their quality updated by a different amount, you'd have to keep adding new classes here.

I'm not sure yet if a larger structural change is needed or not, but some things on my mind are:
- The `DecreaseBy2` and `DecreaseToZero` both have an `update` method, but it's used in different ways. In one, you give it the value to subtract from, and in the other, you give it the value, but then do nothing with it and just return 0. That's a few more code-smells.
  - Implementations of a method shouldn't act that differently from one another
  - You don't want to hand a class/method an argument that goes unused. That tells me there's something off with this pattern / it's not the right abstraction.
- Seeing that all these classes that implement `Updatable` need to know about the idea of `quality` makes me think that the internals of the `Item` class are getting too spread out in your system. The idea of quality now jumps from Item > GildedRose > AgedBrie (or any other similar class) > Updatable > DefaultIncrease (or any other class that implements Updatable)

Reply:
+1

Reply:
>The DecreaseBy2 and DecreaseToZero both have an update method, but it's used in different ways. In one, you give it the value to subtract from, and in the other, you give it the value, but then do nothing with it and just return 0. That's a few more code-smells.

When this was implemented my spidey senses did activate for a split second that something was off. I just didn't think to much into it. But this is good feedback, and something I can take moving forward regarding the implementations of a method.

As Im typing this out I feel like perhaps the abstraction could have been `sellInBehavior` and `qualityBehavior` since those are really what we are looking at. The fact that it is `updateable` doesn't really give the next developer much. But if it were `sellInBehavior` and then the `update` method will describe the update behavior of sellIn. (if we were to stay on the path of strategy pattern)

Reply:
Also with this being a blocker. What is the correct approach here? Do I ask, what does unblocking this look like for you?

Reply:
I think just doing what Steven suggested above:

> if this arithmetic could just live in the implementations of updateQuality and updateSellIn in each of the respective Item classes.

I'd remove all the `DecreaseBy2` -type classes and put that logic back in the respective item classes.

Reply:
This makes sense to us, we removed the inheritance (extends `DefaultItem`) from every special class (except for `Conjured`) and removed all of the `DecreaseBy2` - type classes and put that logic back in the respective item classes.

The reason we kept the `Conjured extends DefaultItem` is because the functionality of the quality degradation of a `ConjuredItem` is inherently linked/coupled with that of the DefaultItem. This being "`Conjured` items degrade in Quality twice as fast as normal items".

This blocker solution is addressed from [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/16/commits/787edac2b69be3565984ac07747d058edd313cba) to [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/16/commits/b4a7a025d8519c2f1355aff41986085493192ee6) (small commits so the whole solution spanned across multiple)